kedify-proxy/templates/envoy-deployment.yaml

```diff
@@ -0,0 +1,105 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "kedify-proxy.fullname" . }}
+  labels:
+    {{- include "kedify-proxy.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.deployment.replicas }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "kedify-proxy.selectorLabels" . | nindent 6 }}
+  {{- with .Values.deployment.rollingUpdate }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      {{- toYaml . | nindent 6 }}
+  {{- end }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "kedify-proxy.labels" . | nindent 8 }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      containers:
+        - name: envoy
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          args:
+            - "--log-format"
+            - {"Timestamp":"%Y-%m-%dT%T.%F","ThreadId":"%t","SourceLine":"%s:%#","Level":"%l","Message":"%j"}
```

Comment:
@wozniakjan since we recently discussed logging for all the components (ideally structured/json), can we add some flexbiility here?

Reply:
attaching docs to the current implementation, users can switch between structured and default plaintext with `KEDIFY_PROXY_LOG_FORMAT` https://kedify.io/documentation/concepts/kedify-agent/#environment-variables

and fyi @jkremser I am currently looking into envoy access logs, so there will likely be another config knob soon for those. No need to add that yet, but sharing this just so you have a full context.
https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage

Reply:
ok, so we might want to add a `logging` section to the values file for the sake of consistency with the agent and KEDA

Reply:
should be done in the latest version, however it looks a bit odd, because there is only 1 config option. It's not our code so i guess it doesn't have the same switches as golang app with zap logger so it's not 1:1 match, but I aligned it